File: app/src/main/java/com/example/android/miwok/Word.java
package com.example.android.miwok;

/**
 * {@link Word} represents a vocabulary word that the user wants to learn.
 * It contains a default translation and a Miwok translation for that word.
 */
public class Word {

    /**
     * Default translation for the word
     */
    private String defaultTranslation;

    /**
     * Miwok translation for the word
     */
    private String miwokTranslation;

    /**
     * Image resource ID for de word
     */
    private int imageResourceId = NO_IMAGE_PROVIDED;

    /**
     * Constant value that represents no image was provided for this word
     */
    private static final int NO_IMAGE_PROVIDED = -1;

    /**
     * Audio resource ID for the word
     */
    private int audioResourceId;

    /**
     * @param defaultTranslation is the word in a language that the user is already familiar with
     *                           (such as English)
     * @param miwokTranslation   is the word in the Miwok language
     */
    public Word(String defaultTranslation, String miwokTranslation, int audioResourceId) {
        this.defaultTranslation = defaultTranslation;
        this.miwokTranslation = miwokTranslation;
        this.audioResourceId = audioResourceId;
    }

    /**
     * @param defaultTranslation is the word in a language that the user is already familiar with
     *                           (such as English)
     * @param miwokTranslation   is the word in the Miwok language
     * @param imageResourceId    is the drawable resource ID for the image associated with the word
     */
    public Word(String defaultTranslation, String miwokTranslation, int imageResourceId, int audioResourceId) {
        this.defaultTranslation = defaultTranslation;
        this.miwokTranslation = miwokTranslation;
        this.imageResourceId = imageResourceId;
        this.audioResourceId = audioResourceId;
    }

    /**
     * Get the default translation of the word.
     */
    public String getDefaultTranslation() {
        return defaultTranslation;
    }

    /**
     * Get the Miwok translation of the word.
     */
    public String getMiwokTranslation() {
        return miwokTranslation;
    }

    /** Return the image resource ID for the word. */
    public int getImageResourceId() {
        return imageResourceId;
    }

    /**
     * Returns whether or not there is an image for this word
     */
    public boolean hasImage() {
        return imageResourceId != NO_IMAGE_PROVIDED;
    }
}

File: app/src/main/java/com/example/android/miwok/Word.kt
package com.example.android.miwok

/**
 * [Word] represents a vocabulary word that the user wants to learn.
 * It contains a default translation and a Miwok translation for that word.
 */
class Word {

    /**
     * Default translation for the word
     */
    var defaultTranslation: String
        private set

    /**
     * Miwok translation for the word
     */
    var miwokTranslation: String
        private set

    /**
     * Image resource ID for de word
     */
    var imageResourceId = NO_IMAGE_PROVIDED
        private set

    /**
     * Audio resource ID for the word
     */
    var audioResourceId: Int
        private set

    /**
     * @param defaultTranslation is the word in a language that the user is already familiar with
     * (such as English)
     * @param miwokTranslation   is the word in the Miwok language
     */
    constructor(defaultTranslation: String, miwokTranslation: String, audioResourceId: Int) {
        this.defaultTranslation = defaultTranslation
        this.miwokTranslation = miwokTranslation
        this.audioResourceId = audioResourceId
    }

    /**
     * @param defaultTranslation is the word in a language that the user is already familiar with
     * (such as English)
     * @param miwokTranslation   is the word in the Miwok language
     * @param imageResourceId    is the drawable resource ID for the image associated with the word
     */
    constructor(
        defaultTranslation: String,
        miwokTranslation: String,
        imageResourceId: Int,
        audioResourceId: Int
    ) {
        this.defaultTranslation = defaultTranslation
        this.miwokTranslation = miwokTranslation
        this.imageResourceId = imageResourceId
        this.audioResourceId = audioResourceId
    }

    /**
     * Returns whether or not there is an image for this word
     */
    fun hasImage(): Boolean {
        return imageResourceId != NO_IMAGE_PROVIDED
    }

    companion object {
        /**
         * Constant value that represents no image was provided for this word
         */
        private const val NO_IMAGE_PROVIDED = -1
    }
}